### DISCUSSION TITLE
thread/thread-state-and-method

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#114] [Announcements]
# thread/thread-state-and-method

Java线程的6种状态及切换(透彻讲解)

https://tobebetterjavaer.com/thread/thread-state-and-method.html

**Comment by tuzkiruncode** [comment]
*线程中断* 这段判断中断状态两个方法是不是有点问题？
应该是这样吧：静态方法是 interrupted()，实例方法是 isInterrupted()。然后静态方法 interrupted() 会影响中断状态，它调用一次就会reset中断状态，但是返回值是reset前的状态。java doc中“In other words, if this method were to be called twice in succession, the second call would return false”，这个描述的意思是，第一次调用的时候，中断状态会被设置为false，所以第二次调用的时候肯定会返回false（不考虑两次调用有其他影响中断状态的操作）。
老师你是不是把实例方法中的参数（boolean ClearInterrupted）跟线程的中断状态整混了？
